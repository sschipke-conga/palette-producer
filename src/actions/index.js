

export const setUser = user => ({
  type: 'SET_USER',
  user
});

export const setAllProjects = allProjects => ({
  type: 'SET_PROJECTS',
  allProjects
});

export const setCurrentPalette = palette => ({
  type: 'SET_CURRENT_PALETTE',
  palette
});

export const setCurrentPaletteInfo = info => ({
  type: 'SET_CURRENT_PALETTE_INFO',
  info
});

export const setCurrentProjectInfo = info => ({
  type: 'SET_CURRENT_PROJECT_INFO',
  info
});

export const setAllPalettes = allPalettes => ({
  type: 'SET_PALETTES',
  allPalettes
});

export const toggleMenu = () => ({
  type: 'TOGGLE_MENU'
})

export const removePalette = (id) => ({
  type: 'REMOVE_PALETTE',
  id: parseInt(id)
});