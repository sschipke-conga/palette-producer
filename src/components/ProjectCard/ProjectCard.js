import './ProjectCard.scss';
import React from 'react';
import PaletteCard from '../PaletteCard/PaletteCard';
import PropTypes from 'prop-types';
import { IoIosAddCircleOutline, IoIosCloseCircleOutline } from 'react-icons/io';
import ProjectContainer from '../ProjectContainer/ProjectContainer';

export const ProjectCard = ({ project, select, palettes, removePalette, removeProject }) => {
  let displayPalettes = 'Add some palettes'
  if (palettes) {
    displayPalettes = palettes.map((palette, index) => {
      return (
        <PaletteCard
          key={'PaletteCard' + index}
          project={project}
          select={select}
          palette={palette}
          removePalette={removePalette}
        />
      )
    })
  }

  return (
    <div className='ProjectCard' id={project.d} name={project.name}>
      <h3 className="projectCard-header-name">{project.name}</h3>
      <div className='add-palette-container' onClick={() => select(project, { name: '' })}>
        <IoIosAddCircleOutline className='add-palette-icon' />
      </div>
      {displayPalettes}
      <IoIosCloseCircleOutline className='delete-project'
        id={project.id}
        onClick={removeProject} />
    </div >
  )
}

export default ProjectCard;

ProjectCard.propTypes = {

}
