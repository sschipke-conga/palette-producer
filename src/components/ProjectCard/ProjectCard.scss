.ProjectCard {
  align-items: center;
  border: 2px solid gray;
  display: flex;
  flex-wrap: nowrap;
  height: 400px;
  margin: auto;
  overflow-x: auto;
  width: 90%;
}

.add-palette-container {
  align-items: center;
  background: rgb(244, 244, 244);
  border: 2px black solid;
  cursor: pointer;
  display: flex;
  height: 200px;
  justify-content: center;
  margin-left: 20px;
  width: 200px;
  &:hover {
    background: rgb(194, 194, 194);
  } 
  &:active {
    background: rgb(177, 177, 177);
  }
}

.add-palette-icon {
  height: 50px;
  width: 50px;
}

.projectCard-header-name {
  width: 150px;
}

.delete-project {
  background: rgb(244, 244, 244);
  border-radius: 5px;
  cursor: pointer;
  height: 50px;
  margin-left: 22px;
  width: 50px;
  &:hover {
    opacity: 0.5;
  }
}