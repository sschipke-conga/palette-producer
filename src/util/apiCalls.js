export const createNewUser = async user => {
  let url = `${process.env.REACT_APP_PALETTE_PRODUCER_BACKEND_BASE_URL}/api/v1/signup`;
  let options = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  let res = await fetch(url, options);

  if (res.status === 500) {
    throw Error("This username already taken");
  }

  if (!res.ok) {
    throw Error("Woops! Something went wrong");
  }

  return res.json();
};

export const loginUser = async user => {
  let url = `${process.env.REACT_APP_PALETTE_PRODUCER_BACKEND_BASE_URL}/api/v1/login`;
  let options = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  let res = await fetch(url, options);

  if (res.status === 401) {
    throw Error("Username or password is incorrect");
  }

  if (!res.ok) {
    throw Error("Woops! Something went wrong");
  }

  return res.json();
};
