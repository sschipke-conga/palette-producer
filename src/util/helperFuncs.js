export const generateRandomHex = () => {
  let hexCode = "#" + Math.floor(Math.random() * 16777215).toString(16);
  if (hexCode.length < 7) {
    return generateRandomHex();
  }
  return hexCode;
};
