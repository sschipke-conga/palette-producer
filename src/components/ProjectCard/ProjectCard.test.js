import React from "react";
import { ProjectCard, mapStateToProps, mapDispatchToProps} from "./ProjectCard";
import { shallow } from "enzyme";
import {deleteProject} from '../../util/apiCalls'
import {removeProject} from '../../actions'
jest.mock('../../util/apiCalls')
const mockRemoveProject = jest.fn()

describe('ProjectCard', () => {
  const mockProject = {
    name: "Steve",
    id: 4
  }
  const mockPalettes = [
    {
      id: 2,
      project_id: 2,
      color1: "#FFFFFF",
      color2: "#FFFFFF",
      color3: "#FFFFFF",
      color4: "#FFFFFF",
      color5: "#FFFFFF"
    }
  ];
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(< ProjectCard
    project={mockProject}
    palettes={mockPalettes}
    removeProject={mockRemoveProject}
      />)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  describe('Remove project button', () => {
    it('call removeProject and deleteProject with the correct arguments', async () => {
      await wrapper.find('.delete-project').simulate('click')
      expect(deleteProject).toHaveBeenCalledWith(4)
      expect(mockRemoveProject).toHaveBeenCalledWith(4)
    })
  })
  describe('mapStateToProps/mapDispatchToProps', () => {
    it('mapStateToProps the user and the state of the menu', () => {
      const mockUser = { username: 'Dave', id: 4 }
      const mockState = {
        user: mockUser,
      };
      const expected = {
        user: mockUser,
      };
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    });
    it('calls dispatch with removeProject action when it is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = removeProject('REMOVE_PROJECT', 4);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.removeProject('REMOVE_PROJECT', 4);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  })
})