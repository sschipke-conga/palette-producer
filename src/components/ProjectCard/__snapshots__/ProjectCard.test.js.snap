// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard should match the snapshot 1`] = `
<div
  className="ProjectCard"
  id={4}
  name="Steve"
>
  <div
    className="project-div"
  >
    <h3
      className="projectCard-header-name"
    >
      Steve
    </h3>
    <button
      className="delete-project"
      id={4}
      onClick={[Function]}
    >
      Delete this project
    </button>
  </div>
  <Connect(PaletteCard)
    key="PaletteCard0"
    palette={
      Object {
        "color1": "#FFFFFF",
        "color2": "#FFFFFF",
        "color3": "#FFFFFF",
        "color4": "#FFFFFF",
        "color5": "#FFFFFF",
        "id": 2,
        "project_id": 2,
      }
    }
    palettesLeft={1}
    project={
      Object {
        "id": 4,
        "name": "Steve",
      }
    }
  />
</div>
`;
